src/EntityFramework/Core/Common/CommandTrees/ExpressionBuilder/DbExpressionBuilder.cs

```diff
@@ -208,33 +208,38 @@ public static DbGroupExpressionBinding GroupBindAs(this DbExpression input, stri
         /// <param name="argument">The argument over which the aggregate function should be calculated.</param>
         /// <exception cref="T:System.ArgumentNullException">function or argument null.</exception>
         /// <exception cref="T:System.ArgumentException">function is not an aggregate function or has more than one argument, or the result type of argument is not equal or promotable to the parameter type of function.</exception>
-        public static DbFunctionAggregate Aggregate(this EdmFunction function, DbExpression argument)
+        public static DbFunctionAggregate Aggregate(this EdmFunction function, IList<DbExpression> arguments)
         {
             Check.NotNull(function, "function");
-            Check.NotNull(argument, "argument");
+			if (arguments.Count == 0)
+			{
+				throw new ArgumentNullException("arguments");
+			}
 
-            return CreateFunctionAggregate(function, argument, false);
+			return CreateFunctionAggregate(function, arguments, false);
         }
 
         /// <summary>
         /// Creates a new <see cref="T:System.Data.Entity.Core.Common.CommandTrees.DbFunctionAggregate" /> that is applied in a distinct fashion.
         /// </summary>
         /// <returns>A new function aggregate with a reference to the given function and argument. The function aggregate's Distinct property will have the value true.</returns>
         /// <param name="function">The function that defines the aggregate operation.</param>
-        /// <param name="argument">The argument over which the aggregate function should be calculated.</param>
+        /// <param name="argument">The arguments over which the aggregate function should be calculated.</param>
         /// <exception cref="T:System.ArgumentNullException">function or argument is null.</exception>
-        /// <exception cref="T:System.ArgumentException">function is not an aggregate function or has more than one argument, or the result type of argument is not equal or promotable to the parameter type of function.</exception>
-        public static DbFunctionAggregate AggregateDistinct(this EdmFunction function, DbExpression argument)
+        /// <exception cref="T:System.ArgumentException">function is not an aggregate function, or the result type of argument is not equal or promotable to the parameter type of function.</exception>
+        public static DbFunctionAggregate AggregateDistinct(this EdmFunction function, IList<DbExpression> arguments)
         {
             Check.NotNull(function, "function");
-            Check.NotNull(argument, "argument");
+            if (arguments.Count == 0) {
```

Comment:
Use new C# 6 feature called null propagation to avoid null reference exception
```csharp
if (arguments?.Any() == false)
```

Fix code identation